core/src/Kokkos_ReductionIdentity.hpp

```diff
@@ -16,421 +16,71 @@
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static T bor()   // 0, only for integer
-type { static_assert( false, "Missing specialization of
-Kokkos::reduction_identity for custom bor reduction type"); return T(); }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static T band()  // !0, only for integer
-type { static_assert( false, "Missing specialization of
-Kokkos::reduction_identity for custom band reduction type"); return T(); }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static T lor()   // 0, only for integer
-type { static_assert( false, "Missing specialization of
-Kokkos::reduction_identity for custom lor reduction type"); return T(); }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static T land()  // !0, only for integer
-type { static_assert( false, "Missing specialization of
-Kokkos::reduction_identity for custom land reduction type"); return T(); }
-};*/
-
-template <>
-struct reduction_identity<char> {
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char sum() {
-    return static_cast<char>(0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char prod() {
-    return static_cast<char>(1);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char max() { return CHAR_MIN; }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char min() { return CHAR_MAX; }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char bor() {
-    return static_cast<char>(0x0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char band() {
-    return ~static_cast<char>(0x0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char lor() {
-    return static_cast<char>(0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static char land() {
-    return static_cast<char>(1);
-  }
-};
-
-template <>
-struct reduction_identity<signed char> {
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char sum() {
-    return static_cast<signed char>(0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char prod() {
-    return static_cast<signed char>(1);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char max() {
-    return SCHAR_MIN;
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char min() {
-    return SCHAR_MAX;
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char bor() {
-    return static_cast<signed char>(0x0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char band() {
-    return ~static_cast<signed char>(0x0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char lor() {
-    return static_cast<signed char>(0);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static signed char land() {
-    return static_cast<signed char>(1);
-  }
-};
-
-template <>
-struct reduction_identity<bool> {
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool lor() {
-    return static_cast<bool>(false);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool land() {
-    return static_cast<bool>(true);
-  }
+struct reduction_identity;
+
+template <typename Integral>
+  requires(std::integral<Integral>)
```

Comment:
```suggestion
template <std::integral Integral>
```
would be more concise.

Reply:
I am waiting on CI to finish running but I agree with applying that suggestion

Reply:
Looks like this was missed.

Reply:
I judged it was not worth re-running the CI and no other reviewers expressed they wanted that change.